airflow/providers/google/cloud/_internal_client/secret_manager_client.py

```diff
@@ -96,3 +96,10 @@ def get_secret(self, secret_id: str, project_id: str, secret_version: str = 'lat
                 secret_id,
             )
             return None
+        except InvalidArgument:
+            self.log.error(
+                """Google Cloud API Call Error (InvalidArgument): Invalid secret ID %s.
+                Only allow secret ID with English letters (A-Z), numbers (0-9), dashes (-), and underscores (_) """,
+                secret_id,
+            )
+            return None
```

Comment:
```suggestion
        except InvalidArgument:
            self.log.error(
                "Google Cloud API Call Error (InvalidArgument): Invalid secret ID %s. "
                "Only ASCII alphabets (A-Z), numbers (0-9), dashes (-), and underscores (_) "
                "are allowed in the secret ID.",
                secret_id,
            )
            return None
```

(Are only uppercase alphabets allowed, or are lowercases also? The message doesn't currently make that clear.)